contracts/L1/gateway/L1ArbitrumMessenger.sol

```diff
@@ -39,9 +39,32 @@ abstract contract L1ArbitrumMessenger {
         uint256 gasPriceBid,
         bytes memory data
     ) internal returns (uint256) {
-        uint256 seqNum = inbox.createRetryableTicket{value: msg.value}(
+        return
+            sendTxToL2(
+                target,
+                user,
+                msg.value,
+                0, // we always assume that l2CallValue = 0
+                maxSubmissionCost,
+                maxGas,
+                gasPriceBid,
+                data
+            );
+    }
+
+    function sendTxToL2(
+        address target,
+        address user,
```

Comment:
should change this param name to `from` as this could also be a contract address now